src/solvers/sweep_solve.jl

```diff
@@ -1,40 +1,30 @@
 
-region_callback(problem; kws...) = nothing
-
-function sweep_callback(problem; outputlevel, sweep, nsweeps, kws...)
-  if outputlevel >= 1
-    println("Done with sweep $sweep/$nsweeps")
-  end
+function default_region_callback(sweep_iterator; kwargs...)
+  return sweep_iterator
 end
-
+function default_sweep_callback(sweep_iterator; kwargs...)
+  return sweep_iterator
+end
+# In this implementation the function `sweep_solve` is essentially just a wrapper around 
+# the iterate interface that allows one to pass callbacks.
 function sweep_solve(
   sweep_iterator;
+  sweep_callback=default_sweep_callback,
+  region_callback=default_region_callback,
   outputlevel=0,
-  region_callback=region_callback,
-  sweep_callback=sweep_callback,
-  kwargs...,
 )
-  for (sweep, region_iter) in enumerate(sweep_iterator)
-    for (region, region_kwargs) in region_tuples(region_iter)
-      region_callback(
-        problem(region_iter);
-        nsweeps=length(sweep_iterator),
-        outputlevel,
-        region_iterator=region_iter,
-        region,
-        region_kwargs,
-        sweep,
-        kwargs...,
-      )
+  # Don't compute the region iteration automatically as we wish to insert a callback.
+  for _ in PauseAfterIncrement(sweep_iterator)
+    for _ in region_iterator(sweep_iterator)
+      region_callback(sweep_iterator; outputlevel=outputlevel)
```

Comment:
Passing `outputlevel` is a bit funny to me (I think it is there for historical reasons). I would think that it would be easy enough to pass keyword arguments as part of the callback, so I think it can be removed.

Reply:
Otherwise, we could have users pass `sweep_callback_kwargs` and `region_callback_kwargs` NamedTuples.

So basically the alternatives would be:
```julia
sweep_solve(sweep_iterator; region_callback=(x -> default_region_callback(x; outputlevel=1))
```
or:
```julia
sweep_solve(sweep_iterator; region_callback_kwargs=(; outputlevel=1))
```
I'm ok with either one, though I'd bias towards the first one for now since we can always add the second one later, for the sake of keeping this level of the code simpler.

Reply:
I don't agree with passing `outputlevel` as a kwarg, I agree its a bit weird, and I left it in for historical reasons as you mention.

IMO, options like `outputlevel` can be defined as part of a closure in a wrapper function and then exposed to a user, if we (or another user) wishes to do that. We should not assume a notion of `outputlevel` for all callbacks.